### PR TITLE
Added test_lib and migrated more tests from ami-val.

### PR DESCRIPTION
**Pull request by narmaku**

### CODE REVIEW

File: lib/test_lib.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def is_rhel_atomic_host(host):
+    with host.sudo():
+        host.file('/etc/redhat-release').contains('Atomic')
```
Comment (jabia99) [inline lib/test_lib.py:3]: Just curiosity why no return statement?

File: test_suite/cloud/test_aws.py [hunk 2]
```diff
@@ -53,6 +55,102 @@ def test_nouveau_is_blacklisted(self, host):
         assert host.file(file_to_check).contains('blacklist nouveau'), \
             f'nouveau is not blacklisted in "{file_to_check}"'
 
+    def test_unwanted_packages_are_not_present(self, host):
+        unwanted_pkgs = [
```
Comment (jabia99) [inline test_suite/cloud/test_aws.py:59]: I'm guessing some of this packages are also going to be part of a generic test. Or are all of them specific to aws?

File: lib/test_lib.py (hunk 1 above)
Comment (narmaku) [reply lib/test_lib.py:3]: Because I forgot :rofl: good catch!

File: test_suite/cloud/test_aws.py (hunk 2 above)
Comment (narmaku) [reply test_suite/cloud/test_aws.py:59]: AFAIK, they are specific to AWS. However there might be some of them that are common to other clouds.
We won't know until we implement the same tests for other providers.

File: lib/test_lib.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+def is_rhel_atomic_host(host):
+    with host.sudo():
+        return host.file('/etc/redhat-release').contains('Atomic')
+
+
+def is_rhel_sap(host):
```
Comment (jabia99) [inline lib/test_lib.py:6]: we need a return here to, right?

File: lib/test_lib.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    __test_keyword_in_repositories(host, 'sap-bundle')
+
+
+def is_rhel_high_availability(host):
```
Comment (jabia99) [inline lib/test_lib.py:10]: same as above

Comment (narmaku) [reply lib/test_lib.py:10]: True

File: lib/test_lib.py (hunk 3 above)
Comment (narmaku) [reply lib/test_lib.py:6]: True

**Pull request merged**